compact_str/src/repr/mod.rs

```diff
@@ -243,21 +269,45 @@ impl Repr {
         match self.discriminant() {
             Discriminant::Heap => {
                 // SAFETY: We checked the discriminant to make sure the union is `heap`
-                MutStrongRepr::Heap(unsafe { &mut self.heap })
+                MutStrongRepr::Heap(unsafe { &mut self.as_union_mut().heap })
             }
             Discriminant::Inline => {
                 // SAFETY: We checked the discriminant to make sure the union is `inline`
-                MutStrongRepr::Inline(unsafe { &mut self.inline })
+                MutStrongRepr::Inline(unsafe { &mut self.as_union_mut().inline })
             }
         }
     }
+
+    #[inline(always)]
+    const fn from_inline(repr: InlineString) -> Self {
+        // SAFETY: An `InlineString` and `Repr` have the same size and alignment
```

Comment:
Nit: alignment doesn't matter for by-value `tranamute`.

Reply:
Updated!